### PR TITLE
Gherkin and Automation of e2e support for event sinks

### PR DESCRIPTION
**Pull request by sanketpathak**
Epic: https://issues.redhat.com/browse/ODC-6390
Story: https://issues.redhat.com/browse/ODC-6503

Acceptance criteria:

-  Add an Event Sink Catalog to the Console

    1. Include new tile on Add page which drills into the *Event Sink *catalog
    2. Include new Event Sink Catalog
    3. Include new catalog type in Dev Catalog, thus Event Sinks are shown in Developer catalog along with other items
    4. Provide side panel for Event Sinks, similar to Event Source experience
    5. Provide a form driven experience for Event Sinks by default, as we do for Event Sources
    6. Provide the ability to switch between YAML/Form in the Event Sink creation process
    7. Add to project menu in Topology should have an Event Sink option
-  Update Event Source flow to allow for an Event Sink as a valid target, both in form & topology D&D flows
-  Update Add Trigger flow to allow for Event Sink as valid subscriber, both in form & topology D&D flows
-  Update Add Subscription flow to allow for Event Sink as valid subscriber, both in form & topology D&D flows

**Checks required for approving Epic gherkin scripts PR:**

- [x] Add @epic-number to the scenarios or feature file [e.g: @odc-xxx]
- [x] Add @to-do for automation possible scenarios
- [x] Add @regression or @smoke based on the importance of the scenario
- [x] Update the test scenarios count in [Automation status confluence Report](https://docs.jboss.org/display/ODC/Automation+Status+Report)
- [x] Check for the linter issues by executing `yarn run gherkin-lint` on frontend folder [Skip epic number tags related linter issues]

**Fix**: https://issues.redhat.com/browse/ODC-6504

**Test Setup**:

Cluster should be running on local

Check the TAGS in frontend/packages/knative-plugin/integration-tests/cypress.json file be: ""(@knative or @knative-eventing) and (@pre-condition or @smoke or @regression) and not (@manual or @to-do or @un-verified or @broken-test)"",

Command to execute:

Run a cluster locally and execute the command `yarn run dev` and then `npm run test-cypress-knative` simultaneously

Execute the eventing/create-event-sink.feature file

**Browser**
Chrome 99

**Test Execution Screenshot:**

### CODE REVIEW

File: frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            Given user has installed Red Hat Integration - Camel K Operator
+              And user has created or selected namespace "aut-event-sink"
+              And user has created channel "kn-channel"
+              And user is at Add page
+
+
+        @regression @odc-6390
+        Scenario: Create event sink from Add page: KES-01-TC01
+             When user clicks on Event Sink card
+              And user clicks on "Kafka Sink" card
```
Comment (sahil143) [inline frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature:15]: Might be better to use a different Event Sink. In future we will have a sink of kind `KafkaSink` whereas the one you have selected is of kind `Kamelet`

File: frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        Scenario: Connect event sink to Broker: KES-01-TC05
+            Given user created Broker "test-broker"
+              And user created "Log Sink" event sink "event-sink-test" with target as "kn-channel"
+              And user is at Topology page
+             When user right clicks on the Broker "test-broker" to open the context menu
+              And user clicks on the Add Trigger
+              And user clicks on the Subscriber dropdown on the modal
+              And user selects the Subscriber as "event-sink-test"
+              And user clicks on Add button
+             Then user will see connection between Broker "test-broker" and Event Sink "event-sink-test"
```
Comment (sahil143) [inline frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature:68]: This test case is wrong. We don't have event sinks available in the Add Trigger modal.
Once user right clicks on Broker or drag connector and drops it on graph, User will see `Add Event Sink` option in the context menu and after that creation flow is as usual.

**Review by sahil143 — CHANGES_REQUESTED** [review]
> Update Event Source flow to allow for an Event Sink as a valid target, both in form & topology D&D flows

This is changed and will not be implemented

> Update Add Trigger flow to allow for Event Sink as valid subscriber, both in form & topology D&D flows

We wont have Event Sinks available on Add Trigger Modal instead it will be available on Drag connector and context menu of Broker. It's already implemented.

> Update Add Subscription flow to allow for Event Sink as valid subscriber, both in form & topology D&D flows

We wont have Event Sinks available on Add Subscription Modal instead it will be available on Drag connector and context menu of Channel. It's already implemented.

File: frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature (hunk 1 above)
Comment (sanketpathak) [reply frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature]: Updated

File: frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature (hunk 2 above)
Comment (sanketpathak) [reply frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature]: Updated

File: frontend/packages/knative-plugin/integration-tests/support/step-definitions/eventing/create-event-sink.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  'user will see connection between Broker {string} and Event Sink {string}',
+  (brokerName: string, eventSinkName: string) => {
+    cy.log(brokerName, eventSinkName);
+    // Below lines to be uncmmented when the feature is complete
+    //   topologyPage.clickOnNode(brokerName);
+    //   topologySidePane.verify();
+    //   cy.byLegacyTestID(eventSinkName).should('be.visible');
+    //   sidePane.close();
+  },
+);
```
[the remark this answers is not in the document]
Comment (sanketpathak) [reply frontend/packages/knative-plugin/integration-tests/support/step-definitions/eventing/create-event-sink.ts]: Updated

File: frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        @regression @manual @odc-6390
+        Scenario: Create event sink through drag and drop from Broker: KES-01-TC05
+            Given user created Broker "test-broker"
+              And user is at Topology page
+             When user drags and drops connector to empty area in topology
+              And user selects "Event Sink" from context menu
+              And user clicks on "Log Sink" card
+              And user clicks on Create Event Sink
+              And user selects Output Target as "kn-channel"
+              And user enters name as "event-sink-test10"
```
Comment (sahil143) [inline frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature:67]: resource is in context here so Resource dropdown will be disabled with broker selected by default.

File: frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+              And user clicks on Create button
+             Then user will see "event-sink-test10" created in topology
```
Comment (sahil143) [inline frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature:69]: need a similar scenario for channel as well

**Review by sahil143 — CHANGES_REQUESTED** [review]

File: frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature (hunk 5 above)
Comment (sanketpathak) [reply frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature]: Added for channel as well

File: frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature (hunk 4 above)
Comment (sanketpathak) [reply frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature]: Updated

File: frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        Scenario: Create event sink through drag and drop from Broker: KES-01-TC05
+            Given user created Broker "test-broker"
+              And user is at Topology page
+             When user drags and drops connector to empty area in topology
+              And user selects "Event Sink" from context menu
+              And user clicks on "Log Sink" card
+              And user clicks on Create Event Sink
+              And user enters name as "event-sink-test10"
+              And user clicks on Create button
+             Then user will see "event-sink-test10" created in topology
```
Comment (sahil143) [inline frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature:68]: we should check if it's connected to the correct broker or channel here.

**Review by sahil143 — CHANGES_REQUESTED** [review]

File: frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature (hunk 6 above)
Comment (sanketpathak) [reply frontend/packages/knative-plugin/integration-tests/features/eventing/create-event-sink.feature]: Updated

**Comment by jrichter1** [comment]
Tests run green, all looks fine
Screenshot from 2022-04-22 12-28-26

/lgtm

**Review by sahil143 — APPROVED** [review]
/lgtm

**Pull request merged**